embedded-graphics/src/coord.rs

```diff
@@ -2,184 +2,230 @@
-            Coord::new(self.0.max(0), self.1.max(0))
-        }
-
-        /// Remove the sign from a coordinate
-        ///
-        ///
-        /// ```
-        /// # use embedded_graphics::coord::Coord;
-        /// #
-        /// let coord = Coord::new(-5, -10);
-        ///
-        /// assert_eq!(coord.abs(), Coord::new(5, 10));
-        /// ```
-        pub fn abs(&self) -> Self {
-            Coord::new(self.0.abs(), self.1.abs())
-        }
+    /// assert_eq!(coord.abs(), Coord::new(5, 10));
+    /// ```
+    pub fn abs(&self) -> Self {
+        Coord::new(self.0.abs(), self.1.abs())
     }
+}
 
-    impl Add for Coord {
-        type Output = Coord;
+impl Add for Coord {
+    type Output = Coord;
 
-        fn add(self, other: Coord) -> Coord {
-            Coord::new(self.0 + other.0, self.1 + other.1)
-        }
+    fn add(self, other: Coord) -> Coord {
+        Coord::new(self.0 + other.0, self.1 + other.1)
     }
+}
 
-    impl AddAssign for Coord {
-        fn add_assign(&mut self, other: Coord) {
-            self.0 += other.0;
-            self.1 += other.1;
-        }
+impl AddAssign for Coord {
+    fn add_assign(&mut self, other: Coord) {
+        self.0 += other.0;
+        self.1 += other.1;
     }
+}
 
-    impl Sub for Coord {
-        type Output = Coord;
+impl Sub for Coord {
+    type Output = Coord;
 
-        fn sub(self, other: Coord) -> Coord {
-            Coord::new(self.0 - other.0, self.1 - other.1)
-        }
+    fn sub(self, other: Coord) -> Coord {
+        Coord::new(self.0 - other.0, self.1 - other.1)
     }
+}
 
-    impl SubAssign for Coord {
-        fn sub_assign(&mut self, other: Coord) {
-            self.0 -= other.0;
-            self.1 -= other.1;
-        }
+impl SubAssign for Coord {
+    fn sub_assign(&mut self, other: Coord) {
+        self.0 -= other.0;
+        self.1 -= other.1;
     }
+}
 
-    impl Index<usize> for Coord {
-        type Output = CoordPart;
+impl Index<usize> for Coord {
+    type Output = i32;
 
-        fn index(&self, idx: usize) -> &CoordPart {
-            match idx {
-                0 => &self.0,
-                1 => &self.1,
-                _ => panic!("Unreachable index {}", idx),
-            }
+    fn index(&self, idx: usize) -> &i32 {
+        match idx {
+            0 => &self.0,
+            1 => &self.1,
+            _ => panic!("Unreachable index {}", idx),
         }
     }
+}
 
-    impl Neg for Coord {
-        type Output = Coord;
+impl Neg for Coord {
+    type Output = Coord;
 
-        fn neg(self) -> Self::Output {
-            Coord::new(-self.0, -self.1)
-        }
+    fn neg(self) -> Self::Output {
+        Coord::new(-self.0, -self.1)
     }
+}
 
-    impl From<(u32, u32)> for Coord {
-        fn from(other: (u32, u32)) -> Self {
-            Self(other.0 as i32, other.1 as i32)
-        }
+impl From<(u32, u32)> for Coord {
+    fn from(other: (u32, u32)) -> Self {
+        Self(other.0 as i32, other.1 as i32)
```

Comment:
What happens if someone calls this with `u32::max_value()`?
There are a lot of `From<...>` implementations, are all of these really necessary?

Reply:
Excellent point. None of these `u32` conversions should've been implemented for `Coord` in the first place. Might've just been some bad copypasta. At any rate, I've removed all `u32`-based conversions for `Coord`.